comp/trace/agent/agent.go

```diff
@@ -0,0 +1,60 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package agent
+
+import (
+	"context"
+
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/trace/config"
+	pkgagent "github.com/DataDog/datadog-agent/pkg/trace/agent"
+	"github.com/DataDog/datadog-agent/pkg/trace/telemetry"
+)
+
+type dependencies struct {
+	fx.In
+
+	Lc         fx.Lifecycle
+	Shutdowner fx.Shutdowner
+
+	Ctx                context.Context
```

Comment:
Why do you need a context here? This feels like it should be handle by FX lifecycle for everything related to timeout and start/stop signal.

Reply:
As it's a wrapper around pkg/trace, I didn't wanted to replace yet how stopping is handled by the tracer agent. It uses context cancellation to stop.

I'll take a second look to it, to make sure if it's viable to do in this first PR.

Reply:
Your comp could create the context and cancel it when `Stop` is called by the FX lifecycle. Having a component depending on a context is a bit weird.